pkg/image/validation.go

```diff
@@ -0,0 +1,153 @@
+package image
+
+import (
+	"fmt"
+	"strings"
+)
+
+func validate(definition *Definition) error {
+	err := validateAPIVersion(definition)
+	if err != nil {
+		return fmt.Errorf("error validating apiVersion: %w", err)
+	}
+	err = validateImage(definition)
+	if err != nil {
+		return fmt.Errorf("error validating image: %w", err)
+	}
+	err = validateOperatingSystem(definition)
+	if err != nil {
+		return fmt.Errorf("error validating operating system: %w", err)
+	}
+
+	return nil
+}
+
+func validateAPIVersion(definition *Definition) error {
+	// Open to discussion regarding how to validate the else if
+	if definition.APIVersion == (Definition{}.APIVersion) {
+		return fmt.Errorf("apiVersion not defined")
+	} else if len(definition.APIVersion) < 2 {
+		return fmt.Errorf("invalid apiVersion")
+	}
+	return nil
+}
+
+func validateImage(definition *Definition) error {
+	if definition.Image == (Image{}) {
+		return fmt.Errorf("image not defined")
+	}
+	if definition.Image.ImageType == (Definition{}.Image.ImageType) {
+		return fmt.Errorf("imageType not defined")
+	} else if definition.Image.ImageType != "iso" && definition.Image.ImageType != "raw" {
+		return fmt.Errorf("invalid imageType, should be 'iso' or 'raw'")
+	}
+	if definition.Image.BaseImage == (Definition{}.Image.BaseImage) {
+		return fmt.Errorf("baseImage not defined")
+	}
+	if definition.Image.OutputImageName == (Definition{}.Image.OutputImageName) {
+		return fmt.Errorf("outputImageName not defined")
+	}
+	return nil
+}
+
+func validateOperatingSystem(definition *Definition) error {
+	if checkIfOperatingSystemDefined(&definition.OperatingSystem) {
+		return nil
+	}
+	err := validateKernalArgs(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating kernal args: %w", err)
+	}
+	err = validateSystemd(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating systemd args: %w", err)
+	}
+	err = validateUsers(&definition.OperatingSystem)
+	if err != nil {
+		return fmt.Errorf("error validating users: %w", err)
+	}
+	return nil
+}
+
+func checkIfOperatingSystemDefined(os *OperatingSystem) bool {
+	return len(os.KernelArgs) == 0 &&
+		len(os.Users) == 0 &&
+		len(os.Systemd.Enable) == 0 && len(os.Systemd.Disable) == 0 &&
```

Comment:
```suggestion
	os.Systemd == (Systemd{}) &&
```

Reply:
For the Systemd comparison, there's this error with the suggested implementation: 
`Invalid operation: os.Systemd == (Systemd{}) (the operator == is not defined on Systemd)` 
So I'll keep the current implementation for now until a better option comes along.

Reply:
Ah, yeah.. because slices are not comparable directly with each other and `Enable` and `Disable` are both slices.